### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by thomashoneyman**
This PR migrates web-dom-xpath to use GitHub Actions instead of Travis for CI, as tracked for all core libraries in https://github.com/purescript/purescript/issues/3962. It also updates the installation instructions to refer to [Spago](https://github.com/purescript/spago) instead of Bower and adds a Pursuit link.

### CODE REVIEW

**Comment by JordanMartinez** [comment]
CI is failing because [pulp hasn't been updated to v15.0.0](https://github.com/purescript-web/purescript-web-dom-xpath/blob/master/package.json#L14)

**Comment by thomashoneyman** [comment]
@bbarker Do you recognize the current error in CI?

```
* Running tests...
/home/runner/work/purescript-web-dom-xpath/purescript-web-dom-xpath/output/Web.DOM.Document.XPath.ResultType/foreign.js:3
exports.any_type = XPathResult.ANY_TYPE;
                   ^

ReferenceError: XPathResult is not defined
    at Object.<anonymous> (/home/runner/work/purescript-web-dom-xpath/purescript-web-dom-xpath/output/Web.DOM.Document.XPath.ResultType/foreign.js:3:20)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/runner/work/purescript-web-dom-xpath/purescript-web-dom-xpath/output/Web.DOM.Document.XPath.ResultType/index.js:3:16)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
* ERROR: Subcommand terminated with exit code 1
```

**Comment by bbarker** [comment]
I'll take a look soon, though off the top of my head, these tests require a
DOM - could that be the issue?

**Comment by thomashoneyman** [comment]
I'm trying to think of what might be different between Travis (where this build seems to have previously succeeded) and GitHub Actions on Node 10 (where this build seems to be failing). Nothing about the library internals has changed.

**Comment by kl0tl** [comment]
I think Travis wasn’t actually running the tests?

```
  - npm install -g bower	
  - npm install	
script:	
  - bower install	
  - npm run -s build
```

We could fix this by renaming test/index.js to test/browser.js, adding a test/node.js file:

```js
const { JSDOM } = require("jsdom");
global.XPathResult = new JSDOM().window.XPathResult;
require("../output/Test.Main/index.js").main();
```

updating the test script:

```sh
bower install && eslint src && pulp build --main "Test.Main" -I "test" -- --censor-lib --strict && node test/node.js
```

and adding [`jsdom`](https://github.com/jsdom/jsdom) to the development dependencies of the library.

Perhaps we don’t even need `jsdom` through, we could also inline XPathResult definition in test/node.js:

```js
{
  ANY_TYPE: 0,
  NUMBER_TYPE: 1,
  STRING_TYPE: 2,
  BOOLEAN_TYPE: 3,
  UNORDERED_NODE_ITERATOR_TYPE: 4,
  ORDERED_NODE_ITERATOR_TYPE: 5,
  UNORDERED_NODE_SNAPSHOT_TYPE: 6,
  ORDERED_NODE_SNAPSHOT_TYPE: 7,
  ANY_UNORDERED_NODE_TYPE: 8,
  FIRST_ORDERED_NODE_TYPE: 9
}
```

Would you be on board with that?

**Comment by bbarker** [comment]
Today I discovered this repository has a `.travis.yml` ... I must have copied it in from another project as a template. As far as I can recall, I never had any tests running under Travis.

As far as @kl0tl 's suggestions, I assume the underlying DOM API is rather stable, so if it is more expedient to just copy in the relevant definitions to the tests, that sounds good to me. I don't have much experience with jsdom to say how much of an improvement that would offer.

**Comment by JordanMartinez** [comment]
> As far as @kl0tl 's suggestions, I assume the underlying DOM API is rather stable, so if it is more expedient to just copy in the relevant definitions to the tests, that sounds good to me. I don't have much experience with jsdom to say how much of an improvement that would offer.

I think I'm on board with this.

**Comment by thomashoneyman** [comment]
Given that [this implementation in the spec hasn't changed since its introduction in 2004](https://www.w3.org/TR/DOM-Level-3-XPath/xpath.html#XPathResult) I'm on board with hard-coding these values rather than introduce `jsdom`. However, I'm not strictly against `jsdom`, either, it just seems like the higher-maintenance option.

@kl0tl with this in mind, would you mind taking charge of deciding which course of action to take and updating these tests?

**Comment by kl0tl** [comment]
I tried to inline XPathResult definition but this library depends on web-dom-parser, which needs a global DOMParser, so there’s not much point in trying to avoid jsdom.

It’s not perfect though, some tests need a real (or at least better) DOM implementation so I had to turn them off in Node.js. I think running at least some of the test suite has value but this is somewhat awkward 😅 

Feel free to revert my commit if this is not satisfying.

**Comment by thomashoneyman** [comment]
No, the changes look good to me. I'm not able to approve my own pull request, but unless anyone else has an issue with the new implementation I think this can be stamped and merged by anyone with permissions.

**Comment by bbarker** [comment]
Considering none of the tests ran in CI before, I think it's an improvement! At least, it looks like all the tests should still be runnable manually from a browser.

👍

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-  - wget -O $HOME/purescript.tar.gz https://github.com/purescript/purescript/releases/download/$TAG/linux64.tar.gz
-  - tar -xvf $HOME/purescript.tar.gz -C $HOME/
-  - chmod a+x $HOME/purescript
-  - npm install -g bower
-  - npm install
-script:
-  - bower install
-  - npm run -s build
-after_success:
-- >-
```
Comment (JordanMartinez) [inline .travis.yml:18]: No tests were run here...

Comment (thomashoneyman) [reply .travis.yml:18]: @kl0tl enabled the tests via GitHub actions in https://github.com/purescript-web/purescript-web-dom-xpath/pull/6/commits/ada7d6e11e622fc740a67d6fdd6ae41806b6bf08

**Pull request merged**